lang/cs/Org.Apache.REEF.IMRU/API/IMRUJobDefinition.cs

```diff
@@ -27,23 +27,139 @@ namespace Org.Apache.REEF.IMRU.API
+        private readonly int _numberOfMappers;
 
-        internal IMRUJobDefinition(IConfiguration configuration, string jobName)
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="mapFunctionConfiguration">Map function configuration</param>
+        /// <param name="mapInputCodecConfiguration">Map input codec configuration</param>
+        /// <param name="updateFunctionCodecsConfiguration">codec configuration for update 
+        /// function. It is union of TMapInput, TMapOutput and TResult configuration</param>
+        /// <param name="reduceFunctionConfiguration">Reduce function configuration</param>
+        /// <param name="updateFunctionConfiguration">Update function configuration</param>
+        /// <param name="mapOutputPipelineDataConverterConfiguration">Configuration of 
+        /// PipelineDataConverter for TMapOutput</param>
+        /// <param name="mapInputPipelineDataConverterConfiguration">Configuration of 
+        /// PipelineDataConverter for TMapInput</param>
+        /// <param name="partitionedDatasetConfiguration">Configuration of partitioned 
+        /// dataset</param>
+        /// <param name="numberOfMappers">Number of mappers</param>
+        /// <param name="jobName">Job name</param>
+        internal IMRUJobDefinition(
+            IConfiguration mapFunctionConfiguration,
+            IConfiguration mapInputCodecConfiguration,
+            IConfiguration updateFunctionCodecsConfiguration,
+            IConfiguration reduceFunctionConfiguration,
+            IConfiguration updateFunctionConfiguration,
+            IConfiguration mapOutputPipelineDataConverterConfiguration,
+            IConfiguration mapInputPipelineDataConverterConfiguration,
+            IConfiguration partitionedDatasetConfiguration,
+            int numberOfMappers,
+            string jobName)
         {
-            _configuration = configuration;
+            _mapFunctionConfiguration = mapFunctionConfiguration;
+            _mapInputCodecConfiguration = mapInputCodecConfiguration;
+            _updateFunctionCodecsConfiguration = updateFunctionCodecsConfiguration;
+            _reduceFunctionConfiguration = reduceFunctionConfiguration;
+            _updateFunctionConfiguration = updateFunctionConfiguration;
+            _mapOutputPipelineDataConverterConfiguration = mapOutputPipelineDataConverterConfiguration;
+            _mapInputPipelineDataConverterConfiguration = mapInputPipelineDataConverterConfiguration;
+            _partitionedDatasetConfiguration = partitionedDatasetConfiguration;
+            _numberOfMappers = numberOfMappers;
             _jobName = jobName;
         }
 
-        internal IConfiguration Configuration
+        /// <summary>
+        /// Name of the job
+        /// </summary>
+        internal string JobName
         {
-            get { return _configuration; }
+            get { return _jobName; }
         }
 
-        internal string JobName
```

Comment:
I assume you are talking about IMRU JOB definition builder and not IMRU JObDeifniton since comment is under job definiton